### PR TITLE
chore: add storefront for auction houses

### PR DESCRIPTION
**Pull request by joze144**
# Changeset

- Support setting up storefront when creating auction house

### CODE REVIEW

File: packages/core/src/mirage-transactions/store-front.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  storeConfig?: string;
+}
+interface MarketplaceAuctionHousePayload {
+  address: string;
+}
+
+export interface MarktplaceSettingsPayload {
+  meta: MarktetplaceMetaPayload;
+  theme: MarketplaceThemePayload;
+  creators: MarketplaceCreatorPayload[];
```
Comment (codebender828) [inline packages/core/src/mirage-transactions/store-front.ts:42]: Do we need all these parameters?

File: packages/core/src/mirage-transactions/store-front.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    ...storeFrontOptions,
+    address,
+    creators: [{ address: authority.toBase58() }],
+    auctionHouses,
+  };
+};
+
+export const createOrUpdateStoreFront = async (authority: PublicKey, feePayer: PublicKey, settingsUri: string) => {
+  const storePubkey = await Store.getPDA(authority);
+  const storeConfigPubkey = await StoreConfig.getPDA(storePubkey);
```
Comment (codebender828) [inline packages/core/src/mirage-transactions/store-front.ts:80]: If I'm not mistaken, I remember that the `Store` and `StoreConfig` are used in v1 of auction_house. Do these have anything to do with the previous version of auction_house?

If you've tested and you're sure they work, then it's all good. Just want to make sure that they are compatible 👍

**Review by codebender828 — COMMENTED** [review]
PR looks mostly good. I had some questions about the `Store` and `StoreConfig` modules. I recall them being from the deprecated version of AuctionHouse.

Just want to make sure you've tested them before merging them.

**Review by codebender828 — APPROVED** [review]

File: packages/core/src/mirage-transactions/store-front.ts (hunk 2 above)
Comment (joze144) [reply packages/core/src/mirage-transactions/store-front.ts:80]: Yes it's tested, it works ok

**Pull request merged**